var/spack/repos/builtin/packages/orca/package.py

```diff
@@ -19,15 +19,21 @@ class Orca(Package):
     homepage = "https://cec.mpg.de"
     url      = "file://{0}/orca_4_0_1_2_linux_x86-64_openmpi202.tar.zst".format(os.getcwd())
 
+    version('4.2.1', sha256='9bbb3bfdca8220b417ee898b27b2885508d8c82799adfa63dde9e72eab49a6b2',
+            expand=False)
+    version('4.2.0', sha256='55a5ca5aaad03396ac5ada2f14b61ffa735fdc2d98355e272465e07a6749d399',
+            expand=False)
     version('4.0.1.2', sha256='cea442aa99ec0d7ffde65014932196b62343f7a6191b4bfc438bfb38c03942f7',
             expand=False)
 
     depends_on('zstd', type='build')
-    depends_on('openmpi@2.0.0:2.1.5', type='run')
+    depends_on('openmpi@2.0.2', type='run', when='@4.0.1.2')
+    depends_on('openmpi@3.1.4', type='run', when='@4.2.0:')
 
     def url_for_version(self, version):
-        out = "file://{0}/orca_{1}_linux_x86-64_openmpi202.tar.zst"
-        return out.format(os.getcwd(), version.underscored)
+        mpiversion = self.spec['openmpi'].version.joined
```

Comment:
Based on the failing tests, I don't think the spec is concretized at this point in time. For example, if you run `spack info orca` it should crash. You'll probably just need to make a dict mapping between the two versions and use it in both `depends_on` and `url_for_version`.

Reply:
Ok, I understand the problem. I will work on something tomorrow.

Reply:
I have repushed this PR with the needed changes. Note that I had to sacrifice the version range because it's harder to map.